File: frontend/src/Types/Recommendation/recommendationDto.ts
import { CriteriaEstimateDto } from '../Criteria/criteriaEstimateDto.ts';

export type RecommendationDto = {
  id: string;
  businessId: string;
  totalScore: number;
  criteriaEstimates: CriteriaEstimateDto[];
  pluses: string[];
  minuses: string[];
  recommendations: string[];
}

File: backend/ReadyBusinesses.Common/Dto/Recommendation/RecommendationDto.cs
using ReadyBusinesses.Common.Dto.Criteria;

namespace ReadyBusinesses.Common.Dto.Recommendation;

public class RecommendationDto
{
    public Guid Id { get; set; }
    
    public Guid BusinessId { get; set; }
    
    public double TotalScore { get; set; }

    public IEnumerable<CriteriaEstimateDto> CriteriaEstimates { get; set; } = [];

    public string[] Pluses { get; set; } = [];

    public string[] Minuses { get; set; } = [];
    
    public string[] Recommendations { get; set; } = [];
}